images/build/slim.Dockerfile

```diff
@@ -0,0 +1,286 @@
+COPY images/receivePgpKeys.sh /tmp/scripts
+RUN chmod +x /tmp/scripts/receivePgpKeys.sh
+RUN set -ex \
+ && . /tmp/scripts/__nodeVersions.sh \
+ && /tmp/scripts/receivePgpKeys.sh 6A010C5166006599AA17F08146C2130DFD2497F5 \
+ && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
+ && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
+ && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
+ && mkdir -p /opt/yarn \
+ && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/yarn \
+ && mv /opt/yarn/yarn-v$YARN_VERSION /opt/yarn/$YARN_VERSION \
+ && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz
+
+RUN set -ex \
+ && . /tmp/scripts/__nodeVersions.sh \
+ && ln -s $NODE8_VERSION /opt/nodejs/$NODE8_MAJOR_MINOR_VERSION \
+ && ln -s $NODE8_MAJOR_MINOR_VERSION /opt/nodejs/8 \
+ && ln -s $NODE10_VERSION /opt/nodejs/$NODE10_MAJOR_MINOR_VERSION \
+ && ln -s $NODE10_MAJOR_MINOR_VERSION /opt/nodejs/10 \
+ && ln -s 10 /opt/nodejs/lts
+RUN set -ex \
+ && ln -s 6.9.0 /opt/npm/6.9 \
+ && ln -s 6.9 /opt/npm/6 \
+ && ln -s 6 /opt/npm/latest
+RUN set -ex \
+ && . /tmp/scripts/__nodeVersions.sh \
+ && ln -s $YARN_VERSION /opt/yarn/stable \
+ && ln -s $YARN_VERSION /opt/yarn/latest \
+ && ln -s $YARN_VERSION /opt/yarn/$YARN_MINOR_VERSION \
+ && ln -s $YARN_MINOR_VERSION /opt/yarn/$YARN_MAJOR_VERSION
+RUN set -ex \
+ && mkdir -p /links \
+ && cp -s /opt/nodejs/lts/bin/* /links \
+ && cp -s /opt/yarn/stable/bin/yarn /opt/yarn/stable/bin/yarnpkg /links
+
+###
+# Python intermediate stages
+# Docker doesn't support variables in `COPY --from`, so we're using intermediate stages
+###
+FROM mcr.microsoft.com/oryx/python-build-base:3.7-${PYTHON_BASE_TAG} AS py37-build-base
+###
+# End Python intermediate stages
+###
+
+FROM main AS python
+# It's not clear whether these are needed at runtime...
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+        tk-dev \
+        uuid-dev \
+ && rm -rf /var/lib/apt/lists/*
+# https://github.com/docker-library/python/issues/147
+ENV PYTHONIOENCODING UTF-8
+COPY build/__pythonVersions.sh /tmp/scripts
+COPY --from=py37-build-base /opt /opt
+RUN . /tmp/scripts/__pythonVersions.sh && set -ex \
+ && [ -d "/opt/python/$PYTHON37_VERSION" ] && echo /opt/python/$PYTHON37_VERSION/lib >> /etc/ld.so.conf.d/python.conf \
+ && ldconfig
+# The link from PYTHON38_VERSION to 3.8.0 exists because "3.8.0b1" isn't a valid SemVer string.
+RUN . /tmp/scripts/__pythonVersions.sh && set -ex \
+ && ln -s $PYTHON37_VERSION /opt/python/latest \
+ && ln -s $PYTHON37_VERSION /opt/python/3.7 \
+ && ln -s 3.7 /opt/python/3
+RUN set -ex \
+ && cd /usr/local/bin \
+ && cp -sn /opt/python/3/bin/* . \
+ # Make sure the alias 'python' always refers to Python 3 by default
+ && ln -sf /opt/python/3/bin/python python
+
+# This stage is used only when building locally
+FROM dotnet-install AS buildscriptbuilder
```

Comment:
FYI: BuildScriptGenerator needs to be part of the slim image (understandably), but the side effect of having this here is that if you are frequently making changes to the BuildScriptGenerator and want to test the change, now the building of the full image would be slow every time as the change in content of the slim image would cause a cache bust resulting in the full build image to be built again.